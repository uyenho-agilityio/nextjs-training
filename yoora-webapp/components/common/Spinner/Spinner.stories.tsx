// Lib
import Storybook from '@storybook/react';

// Component
import { Spinner } from '.';

export default {
  title: 'Components/Spinner',
  component: Spinner,
  argTypes: {
    backgroundColor: { control: 'color' },
  },
} as Storybook.Meta<typeof Spinner>;

const Template: Storybook.StoryFn<typeof Spinner> = args => <Spinner {...args} />;

export const Default = Template.bind({});
