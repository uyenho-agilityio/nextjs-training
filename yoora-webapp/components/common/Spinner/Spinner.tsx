'use client';

// Libs
import {
  Container,
  Spinner as ChakraSpinner,
  SpinnerProps as ChakraSpinnerProps,
} from '@chakra-ui/react';
import React from 'react';

type SpinnerProps = {
  variant?: 'default' | 'center';
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
  color?: string;
  thickness?: string;
} & ChakraSpinnerProps;

export const Spinner: React.FC<SpinnerProps> = ({
  variant = 'default',
  size = 'lg',
  color = 'primary-btn-cl',
  thickness = '3px',
  ...props
}): JSX.Element => {
  return (
    <Container variant={variant}>
      <ChakraSpinner size={size} color={color} thickness={thickness} {...props} />
    </Container>
  );
};
