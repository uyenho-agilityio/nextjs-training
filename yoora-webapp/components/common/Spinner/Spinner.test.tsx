// Lib
import { render } from '@testing-library/react';

// Component
import { Spinner } from './Spinner';

describe('Spinner renders', () => {
  test('should render loading component', () => {
    const { container } = render(<Spinner />);
    expect(container).toMatchSnapshot();
  });
});
