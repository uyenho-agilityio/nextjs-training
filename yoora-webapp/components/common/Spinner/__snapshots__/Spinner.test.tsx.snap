// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner renders should render loading component 1`] = `
<div>
  <div
    class="chakra-container css-0"
  >
    <div
      class="chakra-spinner css-17pvw76"
    >
      <span
        class="css-8b45rq"
      >
        Loading...
      </span>
    </div>
  </div>
</div>
`;
